src/GitPrompt.ps1

```diff
@@ -1052,7 +1052,19 @@ if (!(Test-Path Variable:Global:VcsPromptStatuses)) {
 #>
 function Global:Write-VcsStatus {
     Set-ConsoleMode -ANSI
-    $global:VcsPromptStatuses | ForEach-Object { & $_ }
+    $OFS = ""
+    $sb = [System.Text.StringBuilder]::new(200)
+
+    $global:VcsPromptStatuses | ForEach-Object {
+        $str = & $_
+        if ($str) {
+            # $str is quoted in case we get back an array of strings, quoting it will flatten the array
+            # into a single string and $OFS="" will ensure no spaces between each array item.
+            $sb.Append("$str") > $null
+        }
+    }
+
+    $sb.ToString()
```

Comment:
I may be oversimplifying, but isn't this the same as:

```powershell
$OFS = ""
"$($global:VcsPromptStatuses | ForEach-Object { & $_ })"
```

Reply:
Probably not.  I was following the pattern we'd established in Write-Git*Status methods.  Just thinking of the case where AnsiConsole is disabled - that should write to the host and not return anything (or empty strings) and those should concat to an empty string.  So yeah, that should work.  Want me to change this (and test it)?

Reply:
Sure, I'm all for less code. 99.99+% of the time there's a single status function with a single result.